test/lit/passes/j2cl-merge-itables.wast

```diff
@@ -0,0 +1,208 @@
+;; NOTE: Assertions have been generated by update_lit_checks.py --all-items and should not be edited.
+
+;; RUN: foreach %s %t wasm-opt --closed-world --merge-j2cl-itables -all -S -o - | filecheck %s
+
+;; Shared itable instance.
+(module
+  (rec
+    ;; CHECK:      (rec
+    ;; CHECK-NEXT:  (type $Object (sub (struct (field $vtable (ref $Object.vtable)) (field $itable (ref $Object.itable)))))
+    (type $Object (sub (struct
+      (field $vtable (ref $Object.vtable))
+      (field $itable (ref $Object.itable)))))
+
+    ;; CHECK:       (type $SubObject (sub $Object (struct (field $vtable (ref $SubObject.vtable)) (field $itable (ref $Object.itable)))))
+    (type $SubObject (sub $Object (struct
+      (field $vtable (ref $SubObject.vtable))
+      (field $itable (ref $Object.itable)))))
+
+    ;; CHECK:       (type $2 (func))
+
+    ;; CHECK:       (type $function (func))
+    (type $function (func))
+
+    ;; CHECK:       (type $Object.vtable (sub (struct (field structref))))
+    (type $Object.vtable (sub (struct)))
```

Comment:
Perhaps add a comment here, "A structref field will be added here by the pass". That kind of thing makes it easier to read through the test and find the relevant changes.

Reply:
Or, there could be a single comment at the top of the module explaining the expected changes in overview.